src/main/java/com/flowpowered/noise/module/modifier/Range.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.flowpowered.noise.module.modifier;
+
+import com.flowpowered.noise.exception.NoModuleException;
+import com.flowpowered.noise.module.Module;
+
+/**
+ * A modifier module to map a value from one range to another.
+ * For example: 0.5 (-1 &lt; x &lt; 1) -&gt; 0.75 (0 &lt; x &lt; 1)
+ */
+public class Range extends Module {
+    
+    public static final double DEFAULT_CURRENT_LOWER_BOUND = -1f;
+    public static final double DEFAULT_CURRENT_UPPER_BOUND = 1f;
+    public static final double DEFAULT_NEW_LOWER_BOUND = 0f;
+    public static final double DEFAULT_NEW_UPPER_BOUND = 1f;
+    
+    /* Current lower bound */
+    private double currentLowerBound = DEFAULT_CURRENT_LOWER_BOUND;
+    /* Current upper bound */
+    private double currentUpperBound = DEFAULT_CURRENT_UPPER_BOUND;
+    /* New lower bound */
+    private double newLowerBound = DEFAULT_NEW_LOWER_BOUND;
+    /* New Upper Bound */
+    private double newUpperBound = DEFAULT_NEW_UPPER_BOUND;
+    /* Cache variables */
+    private double scale;
+    private double bias;
+
+    public Range() {
+        super(1);
+    }
+
+    public double getCurrentLowerBound() {
+        return currentLowerBound;
+    }
+
+    // Bounds setters are private to prevent changes to bounds
+    // without recalculating scale/bias
+    private void setCurrentLowerBound(double currentLowerBound) {
+        this.currentLowerBound = currentLowerBound;
+    }
+
+    public double getCurrentUpperBound() {
+        return currentUpperBound;
+    }
+
+    private void setCurrentUpperBound(double currentUpperBound) {
+        this.currentUpperBound = currentUpperBound;
+    }
+
+    public double getNewLowerBound() {
+        return newLowerBound;
+    }
+
+    private void setNewLowerBound(double newLowerBound) {
+        this.newLowerBound = newLowerBound;
+    }
+
+    public double getNewUpperBound() {
+        return newUpperBound;
+    }
+
+    private void setNewUpperBound(double newUpperBound) {
+        this.newUpperBound = newUpperBound;
+    }
+    
+    /*
+     * Calculate the scale and biased to be a applied during Range#getValue(int x, int y, int z)
+     * Should be called when the bounds are modified
+     */
+    private void recalculateScaleBias() {
+    	scale = (getNewUpperBound() - getNewLowerBound()) / 
```

Comment:
This should enforce `currentUpperBound != currentLowerBound` and `newUpperBound != newLowerBound`.